weather_mv/loader_pipeline/sinks.py

```diff
@@ -100,17 +127,26 @@ def _get_band_data(i):
     ds = xr.merge(band_data_list)
     ds.attrs['is_normalized'] = ds_is_normalized_attr
 
+    file_time = ds.data_vars['precipitationCal'].attrs['TIFFTAG_DATETIME']
+
+    start_time = match_datetime(uri, initialization_time)
+    end_time = match_datetime(uri, forecast_time)
+
+    print(f"start_time === {start_time}")
+    print(f"end_time === {end_time}")
+
+    ds.attrs['start_time'] = start_time
+    ds.attrs['end_time'] = end_time
+
     # TODO(#159): Explore ways to capture required metadata using xarray.
-    with rasterio.open(filename) as f:
-        datetime_value_ms = None
-        try:
-            datetime_value_ms = f.tags()[tif_metadata_for_datetime]
-            ds = ds.assign_coords({'time': datetime.datetime.utcfromtimestamp(int(datetime_value_ms) / 1000.0)})
-        except KeyError:
-            raise RuntimeError(f"Invalid datetime metadata of tif: {tif_metadata_for_datetime}.")
-        except ValueError:
-            raise RuntimeError(f"Invalid datetime value in tif's metadata: {datetime_value_ms}.")
-        ds = ds.expand_dims('time')
+    datetime_value_ms = None
+    try:
+        datetime_value_s = datetime.datetime.strptime(file_time, '%Y:%m:%d %H:%M:%S').timestamp()
+        ds = ds.assign_coords({'time': datetime.datetime.utcfromtimestamp(int(datetime_value_s))})
+    except KeyError:
+        raise RuntimeError(f"Invalid datetime metadata of tif: {tif_metadata_for_datetime}.")
+    except ValueError:
+        raise RuntimeError(f"Invalid datetime value in tif's metadata: {datetime_value_ms}.")
 
```

Comment:
This is imerg specific deletion. Should handle this smartly.
We should keep this. Removing `ds = ds.expand_dims('time')` will affect other pipelines.

Reply:
Keeping `ds = ds.expand_dims('time')` causes a Floating point exception in the pipeline.